packages/frontend/src/hooks/cartesianChartConfig/useCartesianChartConfig.ts

```diff
@@ -482,16 +486,79 @@ const useCartesianChartConfig = ({
         ];
     }, [resultsData, sortedDimensions]);
 
+    const isFieldValidTableCalculation = useCallback(
+        (fieldName: string) => {
+            return (
+                tableCalculationsMetadata?.some(
+                    (tc) => tc.name === fieldName || tc.oldName === fieldName,
+                ) ?? false
+            );
+        },
+        [tableCalculationsMetadata],
+    );
+
+    useEffect(() => {
+        if (!tableCalculationsMetadata) return;
+
+        const xFieldTcIndex = tableCalculationsMetadata?.findIndex(
+            (tc) => tc.oldName === dirtyLayout?.xField,
+        );
+
+        if (xFieldTcIndex !== -1) {
+            setDirtyLayout((prev) => {
+                return {
+                    ...prev,
+                    xField: tableCalculationsMetadata[xFieldTcIndex].name,
+                };
+            });
+        }
+    }, [dirtyLayout?.xField, tableCalculationsMetadata]);
+
+    useEffect(() => {
+        if (!tableCalculationsMetadata) return;
+
+        const newYFields = dirtyLayout?.yField?.map((yField) => {
+            const yFieldTcIndex = tableCalculationsMetadata.findIndex(
+                (tc) => tc.oldName === yField,
+            );
+
+            if (yFieldTcIndex !== -1) {
+                return tableCalculationsMetadata[yFieldTcIndex].name;
+            }
+
+            return yField;
+        });
+
+        if (
+            !newYFields ||
+            isEqual(sortBy(dirtyLayout?.yField), sortBy(newYFields))
+        )
+            return;
+
+        setDirtyLayout((prev) => {
+            return {
+                ...prev,
+                yField: newYFields,
+            };
+        });
+    }, [dirtyLayout?.yField, tableCalculationsMetadata]);
+
     // Set fallout layout values
     // https://www.notion.so/lightdash/Default-chart-configurations-5d3001af990d4b6fa990dba4564540f6
     useEffect(() => {
```

Comment:
Wondering if we could handle the table calculation metadata logic all in this `useEffect` so that we don't call `setDirtyLayout` _more than once_ - what do you think @almeidabbm ? 
Also, would recommend adding a comment to make it clear why we're doing it 👍

Reply:
Yeah I will try and do it, I had it out because I didn't want to pollute that hook too much as it is too big, but will find a way to make it readable 💪